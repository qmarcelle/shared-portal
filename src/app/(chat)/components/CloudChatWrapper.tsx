'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
