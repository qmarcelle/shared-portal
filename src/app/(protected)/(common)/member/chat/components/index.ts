/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
