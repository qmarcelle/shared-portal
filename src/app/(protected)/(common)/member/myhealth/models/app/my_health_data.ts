import { PrimaryCareProviderDetails } from '@/app/(protected)/(common)/member/findcare/primaryCareOptions/model/api/primary_care_provider';
import { VisibilityRules } from '@/visibilityEngine/rules';

export interface MyHealthData {
  memberRewards: MemberRewards | null;
  primaryCareProvider: PrimaryCareProviderDetails | null;
  visibilityRules?: VisibilityRules;
}

export interface MemberRewards {
  totalAmountEarned: number;
  totalAmount: number;
  quarterlyMaxPoints: number;
  quarterlyPointsEarned: number;
  isSelfFunded: boolean;
  //To Do: Need to Revisit the logic once we get proper api response in phase 2.
  //quarter: 'Q1' | 'Q2' | 'Q3';
}

export enum Quarter {
  'Q1' = 'JAN - APR',
  'Q2' = 'MAY - AUG',
  'Q3' = 'SEP - DEC',
}
