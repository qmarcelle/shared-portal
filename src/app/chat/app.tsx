'use client';

import { ChatWidget } from './components/core/ChatWidget';
import { ChatContextProvider } from './components/providers/ChatContextProvider';
import type { ChatConfig, GenesysUserData } from './types/types';

// Default config for development - should be overridden in production
const defaultConfig: ChatConfig = {
  token: process.env.NEXT_PUBLIC_CHAT_TOKEN || '',
  endPoint: process.env.NEXT_PUBLIC_CHAT_ENDPOINT || '',
  opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
  userID: '',
  memberFirstname: '',
  memberLastname: '',
  memberId: '',
  groupId: '',
  planId: '',
  planName: '',
  businessHours: {
    isOpen24x7: false,
    days: [],
    timezone: 'America/New_York',
    isCurrentlyOpen: true,
    lastUpdated: Date.now(),
    source: 'default',
  },
};

// Default user data for development - should be overridden in production
const defaultUserData: Partial<GenesysUserData> = {
  config: {
    deploymentId: process.env.NEXT_PUBLIC_GENESYS_DEPLOYMENT_ID || '',
    region: process.env.NEXT_PUBLIC_GENESYS_REGION || '',
  },
  userInfo: {
    firstName: '',
    lastName: '',
  },
};

export default function ChatPage() {
  return (
    <ChatContextProvider config={defaultConfig} userData={defaultUserData}>
      <div className="chat-container">
        <ChatWidget />
      </div>
    </ChatContextProvider>
  );
}
