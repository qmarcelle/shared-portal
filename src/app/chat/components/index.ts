// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
