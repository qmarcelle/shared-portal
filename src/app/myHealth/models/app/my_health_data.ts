import { PrimaryCareProviderDetails } from '@/app/findcare/primaryCareOptions/model/api/primary_care_provider';
import { VisibilityRules } from '@/visibilityEngine/rules';

export interface MyHealthData {
  primaryCareProvider: PrimaryCareProviderDetails | null;
  visibilityRules?: VisibilityRules;
}
