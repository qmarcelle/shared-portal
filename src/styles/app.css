@tailwind utilities;

@layer utilities {
  .text-balance {
    text-wrap: balance;
  }
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

@font-face {
  font-family: 'Univers-45';
  src: url(./UniversLTPro-55Roman.woff2);
}

#blueback {
  background: radial-gradient(
      ellipse farthest-corner at right top,
      #5dc1fd 0%,
      #0000 60%
    ),
    linear-gradient(245deg, #067dac 0%, #00497e 70%);
  padding-bottom: 18%;
}

#logo {
  width: 175px;
  margin: 15px;
  display: inline-block;
}

#blueback #mainSection h1 {
  font-size: 24px;
  margin-top: 5px;
}

#blueback h3 {
  font-size: 20px;
  margin-top: 25px;
}

#marginSection {
  width: 416px;
  margin: 0 auto;
}

#mainSection {
  border: 1px solid #cccccc;
  border-radius: 8px;
  background-color: #ffffff;
  padding: 25px 25px;
  margin-top: 18px;
}

.buttonlink {
  color: #ffffff;
  background-color: unset;
  border: none;
  text-decoration: underline;
  margin-top: 25px;
}

#backButton::before {
  content: url('../../public/assets/icon-link_pointer_white.png');
  transform: rotate(180deg);
  display: inline-block;
  margin-right: 5px;
}

#contactbutton,
#resendbutton,
#choosebutton {
  color: #005eb9;
  font-weight: bold;
}

#confirmbutton {
  margin-top: 30px;
  width: 100%;
  height: 36px;
  background-color: #7faedc;
  border: 1px solid #7faedc;
  color: #ffffff;
  border-radius: 3px;
  opacity: 0.7;
}

#choosebutton {
  margin: 0 auto;
  margin-top: 30px;
  margin-bottom: 39px;
  width: 98%;
}

#contactbutton {
  margin: 0px;
  display: inline-block;
  padding: 0;
}

.headerbutton {
  float: right;
  margin: 21px !important;
}

@media (max-width: 767.9px) {
  #marginSection {
    width: 90%;
  }
}
.priorAuthDetail .success,
.priorAuthDetail .error,
.priorAuthDetail .partialapproval,
.priorAuthDetail .neutral {
  width: -moz-fit-content;
  width: fit-content;
}
.priorAuthDetail .statusDetail {
  margin-left: 70px;
}

.access-code {
  width: 114px;
  height: 63px;
  background-color: #e7f6ff;
}
