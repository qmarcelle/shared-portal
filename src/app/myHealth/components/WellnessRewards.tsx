import { ErrorInfoCard } from '@/components/composite/ErrorInfoCard';
import { AppLink } from '@/components/foundation/AppLink';
import { Button } from '@/components/foundation/Button';
import { Divider } from '@/components/foundation/Divider';
import { ProgressBar } from '@/components/foundation/ProgressBar';
import { RichText } from '@/components/foundation/RichText';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { isChipRewardsINTEligible } from '@/visibilityEngine/computeVisibilityRules';
import { VisibilityRules } from '@/visibilityEngine/rules';
import Image from 'next/image';
import { ReactNode } from 'react';
import { IComponent } from '../../../components/IComponent';
import { Card } from '../../../components/foundation/Card';
import { Column } from '../../../components/foundation/Column';
import { Header } from '../../../components/foundation/Header';
import { externalIcon } from '../../../components/foundation/Icons';
import { MemberRewards } from '../models/app/my_health_data';

interface WellnessRewardsProps extends IComponent {
  color1?: string;
  color2?: string;
  memberRewards: MemberRewards | null;
  icon?: ReactNode;
  linkText?: string;
  visibilityRules?: VisibilityRules;
  isMemRelation?: string;
}

export const WellnessRewards = ({
  className,
  // color2 = '#5DC1FD',
  //color1 = '#f2f2f2',
  linkText = 'View Ways to Earn & Learn more',
  icon = <Image alt="external icon" src={externalIcon} />,
  memberRewards,
  visibilityRules,
  isMemRelation,
}: WellnessRewardsProps) => {
  return (
    <>
      <Card className={className}>
        <>
          <Header type="title-2" text="Wellness Rewards" />
          <Spacer size={25} />
          {memberRewards ? (
            <>
              <Row className="justify-between hidden sm:flex">
                <Column className="w-full">
                  {/* To Do: Need to Revisit the logic once we get proper api response in phase 2.
            {memberRewards.quarter && (
              <TextBox
                text={`${memberRewards.quarter} | ${Quarter[memberRewards.quarter]}`}
              />
            )} */}
                  <Spacer size={5} />
                  {memberRewards.isSelfFunded ? (
                    <RichText
                      type="body-1"
                      spans={[
                        <span className="font-bold" key={0}>
                          You&apos;ve earned{' '}
                        </span>,
                        <span className="font-bold" key={1}>
                          ${`${memberRewards.totalAmountEarned} `}
                        </span>,
                        <span key={3}>
                          {` / $${memberRewards.totalAmount}`}
                        </span>,
                      ]}
                    />
                  ) : (
                    <RichText
                      type="body-1"
                      spans={[
                        <span className="font-bold" key={0}>
                          You&apos;ve earned{' '}
                        </span>,
                        <span className="font-bold" key={1}>
                          {`${memberRewards.quarterlyPointsEarned} `}
                        </span>,
                        <span key={3}>
                          {` / ${memberRewards.quarterlyMaxPoints} Points`}
                        </span>,
                      ]}
                    />
                  )}

                  <Spacer size={16} />
                  {memberRewards.isSelfFunded ? (
                    <ProgressBar
                      height={10}
                      completePercent={
                        (memberRewards.totalAmountEarned /
                          memberRewards.totalAmount) *
                        100
                      }
                    />
                  ) : (
                    <ProgressBar
                      height={10}
                      completePercent={
                        (memberRewards.quarterlyPointsEarned /
                          memberRewards.quarterlyMaxPoints) *
                        100
                      }
                    />
                  )}
                  <Spacer size={12} />
                  <Row className="justify-between">
                    {!memberRewards.isSelfFunded ? (
                      <Column>
                        <TextBox className="body-2" text="100 points = $100" />
                      </Column>
                    ) : (
                      <Spacer size={32} />
                    )}

                    <Column className="items-end">
                      {!memberRewards.isSelfFunded ? (
                        <>
                          <TextBox
                            className="font-bold text-end"
                            text={`${memberRewards.quarterlyMaxPoints} pts`}
                          />
                          <TextBox className="body-2" text="Quarterly Max" />
                        </>
                      ) : (
                        <>
                          <TextBox
                            className="font-bold text-end"
                            text={`$${memberRewards.totalAmount}`}
                          />
                          <TextBox className="body-2" text="Annual Max" />
                        </>
                      )}
                    </Column>
                  </Row>
                  <Spacer size={18} />
                  <Row>
                    {isMemRelation &&
                    isChipRewardsINTEligible(visibilityRules) ? (
                      <AppLink
                        className="body-1"
                        label={linkText}
                        url={`/sso/launch?PartnerSpId=${process.env.NEXT_PUBLIC_IDP_CHIP_REWARDS}`}
                      />
                    ) : (
                      <AppLink
                        className="body-1"
                        label={linkText}
                        url={`/sso/launch?PartnerSpId=${process.env.NEXT_PUBLIC_IDP_CHIP_REWARDS}`}
                      />
                    )}
                    <Image alt="external icon" src={externalIcon} />
                  </Row>
                  <Spacer size={18} />
                  <Divider />
                  <Spacer size={32} />

                  <TextBox
                    className="body-2"
                    text="Wellness Rewards is based on your processed claims.There may be a delay in the Wellness Rewards information updating."
                  />
                </Column>
              </Row>
              <Column className="block sm:hidden">
                <Spacer size={5} />
                <TextBox text="My Points" />
                <Spacer size={16} />
                {memberRewards.isSelfFunded ? (
                  <ProgressBar
                    height={10}
                    completePercent={
                      (memberRewards.totalAmountEarned /
                        memberRewards.totalAmount) *
                      100
                    }
                  />
                ) : (
                  <ProgressBar
                    height={10}
                    completePercent={
                      (memberRewards.quarterlyPointsEarned /
                        memberRewards.quarterlyMaxPoints) *
                      100
                    }
                  />
                )}

                <Spacer size={18} />
                <Row className="items-center">
                  <TextBox text="100 points = $100" />
                  &nbsp;|
                  <AppLink label="View Rewards FAQ" />
                </Row>
                <Spacer size={18} />
                <Button label={linkText} icon={icon} callback={() => null} />
              </Column>
            </>
          ) : (
            <ErrorInfoCard errorText="There was a problem loading your information. Please try refreshing the page or returning to this page later." />
          )}
        </>
      </Card>
      {/* <WellnessInfo
        header="Active Rewards - Employer Provided Reward"
        subHeader="Wellness Rewards"
        bodyText="Complete wellness tasks to earn rewards provided by your employer."
        buttonText="Learn More"
        className="section"
      /> */}
    </>
  );
};
