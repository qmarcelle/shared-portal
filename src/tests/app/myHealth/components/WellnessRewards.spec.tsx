import { WellnessRewards } from '@/app/myHealth/components/WellnessRewards';
import { MemberRewards } from '@/app/myHealth/models/app/my_health_data';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = (data: MemberRewards) => {
  return render(<WellnessRewards className="section" memberRewards={data} />);
};

describe('WellnessRewardsSection', () => {
  it('should render the UI correctly', async () => {
    const data = {
      quarterlyPointsEarned: 70,
      quarterlyMaxPoints: 100,
      totalAmountEarned: 0,
      totalAmount: 100,
    };
    const component = renderUI(data);
    expect(screen.getByText('Wellness Rewards')).toBeVisible();
    expect(screen.getAllByText('My Points').length).toBe(2);
    expect(screen.getAllByText('Earned').length).toBe(2);
    expect(screen.getAllByText('Max').length).toBe(2);
    expect(screen.getAllByText('View Rewards FAQ').length).toBe(2);
    expect(screen.getAllByText('View Ways to Earn').length).toBe(2);
    expect(screen.getAllByText(/You've earned/i).length).toBe(2);
    expect(screen.getAllByText('$0.00').length).toBe(2);
    expect(screen.getAllByText('of $100.00 Quarterly Max').length).toBe(2);

    expect(component).toMatchSnapshot();
  });
});
