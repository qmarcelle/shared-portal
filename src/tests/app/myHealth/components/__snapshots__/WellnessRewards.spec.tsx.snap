// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WellnessRewardsSection should render the UI correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-main section"
      >
        <h2
          class="title-2"
        >
          Wellness Rewards
        </h2>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 25px; min-height: 25px;"
        />
        <div
          class="flex flex-row justify-between hidden sm:flex"
        >
          <div
            class="flex flex-col w-1/2"
          >
            <span
              style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
            />
            <p
              class="body-1"
            >
              My Points
            </p>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
            />
            <div>
              <div
                aria-label="bar chart"
                style="height: 10px; width: 70%;"
              />
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
            />
            <div
              class="flex flex-row justify-between"
            >
              <div
                class="flex flex-col"
              >
                <p
                  class="body-1 font-bold"
                >
                  70 pts
                </p>
                <p
                  class="body-1"
                >
                  Earned
                </p>
              </div>
              <div
                class="flex flex-col items-end"
              >
                <p
                  class="body-1 font-bold text-end"
                >
                  100 pts
                </p>
                <p
                  class="body-1"
                >
                  Max
                </p>
              </div>
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
            />
            <div
              class="flex flex-row items-center"
            >
              <p
                class="body-1"
              >
                100 points = $100
              </p>
               |
              <a
                style="display: block;"
                tabindex="-1"
              >
                <button
                  class="flex flex-row link-container undefined"
                  style="max-width: max-content; height: auto; display: block;"
                  tabindex="0"
                >
                  <p
                    class="link undefined"
                    style="display: block;"
                  >
                    View Rewards FAQ
                  </p>
                </button>
              </a>
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
            />
            <button
              aria-label="View Ways to Earn"
              class="button-text primary text-center flex flex-row justify-center items-center min-w-fit null undefined"
              type="button"
            >
              View Ways to Earn
              <p
                class="ml-2"
              >
                <img
                  alt="external icon"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </p>
            </button>
          </div>
          <div
            class=""
            style="width: 1px; height: auto; background-color: rgb(204, 204, 204);"
          />
          <div
            class="flex flex-col"
          >
            <div
              class="flex flex-col chartSection"
            >
              <div
                class="text-center absolute doughnutTextWellness pl-3 w-48"
              >
                <div
                  class="flex flex-col justify-between pl-3 chartText"
                >
                  <p>
                    You've earned
                  </p>
                  <p
                    class="font-bold"
                  >
                    $0.00
                  </p>
                  <p>
                    of 
                    $100.00
                     Quarterly Max
                  </p>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
              />
              <canvas
                class="spendingSummaryChart"
                height="150"
                role="img"
                width="300"
              />
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
            </div>
          </div>
        </div>
        <div
          class="flex flex-col block sm:hidden"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
          />
          <p
            class="body-1"
          >
            My Points
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <div>
            <div
              aria-label="bar chart"
              style="height: 10px; width: 70%;"
            />
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
          />
          <div
            class="flex flex-row justify-between"
          >
            <div
              class="flex flex-col"
            >
              <p
                class="body-1 font-bold"
              >
                70 pts
              </p>
              <p
                class="body-1"
              >
                Earned
              </p>
            </div>
            <div
              class="flex flex-col"
            >
              <p
                class="body-1 font-bold text-end"
              >
                100 pts
              </p>
              <p
                class="body-1"
              >
                Max
              </p>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <div
            class="flex flex-row items-center"
          >
            <p
              class="body-1"
            >
              100 points = $100
            </p>
             |
            <a
              style="display: block;"
              tabindex="-1"
            >
              <button
                class="flex flex-row link-container undefined"
                style="max-width: max-content; height: auto; display: block;"
                tabindex="0"
              >
                <p
                  class="link undefined"
                  style="display: block;"
                >
                  View Rewards FAQ
                </p>
              </button>
            </a>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <button
            aria-label="View Ways to Earn"
            class="button-text primary text-center flex flex-row justify-center items-center min-w-fit null undefined"
            type="button"
          >
            View Ways to Earn
            <p
              class="ml-2"
            >
              <img
                alt="external icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </p>
          </button>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <div
            class=""
            style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 25px; min-height: 25px;"
          />
          <div
            class="flex flex-col chartSection"
          >
            <div
              class="text-center absolute doughnutTextWellness pl-3 w-48"
            >
              <div
                class="flex flex-col justify-between pl-3 chartText"
              >
                <p>
                  You've earned
                </p>
                <p
                  class="font-bold"
                >
                  $0.00
                </p>
                <p>
                  of 
                  $100.00
                   Quarterly Max
                </p>
              </div>
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
            />
            <canvas
              class="spendingSummaryChart"
              height="150"
              role="img"
              width="300"
            />
            <span
              style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-main section"
    >
      <h2
        class="title-2"
      >
        Wellness Rewards
      </h2>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 25px; min-height: 25px;"
      />
      <div
        class="flex flex-row justify-between hidden sm:flex"
      >
        <div
          class="flex flex-col w-1/2"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
          />
          <p
            class="body-1"
          >
            My Points
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <div>
            <div
              aria-label="bar chart"
              style="height: 10px; width: 70%;"
            />
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
          />
          <div
            class="flex flex-row justify-between"
          >
            <div
              class="flex flex-col"
            >
              <p
                class="body-1 font-bold"
              >
                70 pts
              </p>
              <p
                class="body-1"
              >
                Earned
              </p>
            </div>
            <div
              class="flex flex-col items-end"
            >
              <p
                class="body-1 font-bold text-end"
              >
                100 pts
              </p>
              <p
                class="body-1"
              >
                Max
              </p>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <div
            class="flex flex-row items-center"
          >
            <p
              class="body-1"
            >
              100 points = $100
            </p>
             |
            <a
              style="display: block;"
              tabindex="-1"
            >
              <button
                class="flex flex-row link-container undefined"
                style="max-width: max-content; height: auto; display: block;"
                tabindex="0"
              >
                <p
                  class="link undefined"
                  style="display: block;"
                >
                  View Rewards FAQ
                </p>
              </button>
            </a>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <button
            aria-label="View Ways to Earn"
            class="button-text primary text-center flex flex-row justify-center items-center min-w-fit null undefined"
            type="button"
          >
            View Ways to Earn
            <p
              class="ml-2"
            >
              <img
                alt="external icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </p>
          </button>
        </div>
        <div
          class=""
          style="width: 1px; height: auto; background-color: rgb(204, 204, 204);"
        />
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-col chartSection"
          >
            <div
              class="text-center absolute doughnutTextWellness pl-3 w-48"
            >
              <div
                class="flex flex-col justify-between pl-3 chartText"
              >
                <p>
                  You've earned
                </p>
                <p
                  class="font-bold"
                >
                  $0.00
                </p>
                <p>
                  of 
                  $100.00
                   Quarterly Max
                </p>
              </div>
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
            />
            <canvas
              class="spendingSummaryChart"
              height="150"
              role="img"
              width="300"
            />
            <span
              style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col block sm:hidden"
      >
        <span
          style="display: block; width: 1px; min-width: 1px; height: 5px; min-height: 5px;"
        />
        <p
          class="body-1"
        >
          My Points
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div>
          <div
            aria-label="bar chart"
            style="height: 10px; width: 70%;"
          />
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
        />
        <div
          class="flex flex-row justify-between"
        >
          <div
            class="flex flex-col"
          >
            <p
              class="body-1 font-bold"
            >
              70 pts
            </p>
            <p
              class="body-1"
            >
              Earned
            </p>
          </div>
          <div
            class="flex flex-col"
          >
            <p
              class="body-1 font-bold text-end"
            >
              100 pts
            </p>
            <p
              class="body-1"
            >
              Max
            </p>
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
        />
        <div
          class="flex flex-row items-center"
        >
          <p
            class="body-1"
          >
            100 points = $100
          </p>
           |
          <a
            style="display: block;"
            tabindex="-1"
          >
            <button
              class="flex flex-row link-container undefined"
              style="max-width: max-content; height: auto; display: block;"
              tabindex="0"
            >
              <p
                class="link undefined"
                style="display: block;"
              >
                View Rewards FAQ
              </p>
            </button>
          </a>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
        />
        <button
          aria-label="View Ways to Earn"
          class="button-text primary text-center flex flex-row justify-center items-center min-w-fit null undefined"
          type="button"
        >
          View Ways to Earn
          <p
            class="ml-2"
          >
            <img
              alt="external icon"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </p>
        </button>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <div
          class=""
          style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
        />
        <span
          style="display: block; width: 1px; min-width: 1px; height: 25px; min-height: 25px;"
        />
        <div
          class="flex flex-col chartSection"
        >
          <div
            class="text-center absolute doughnutTextWellness pl-3 w-48"
          >
            <div
              class="flex flex-col justify-between pl-3 chartText"
            >
              <p>
                You've earned
              </p>
              <p
                class="font-bold"
              >
                $0.00
              </p>
              <p>
                of 
                $100.00
                 Quarterly Max
              </p>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
          />
          <canvas
            class="spendingSummaryChart"
            height="150"
            role="img"
            width="300"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
